### PR TITLE
Remove enable-taint-manager and pod-eviction-timeout CLI flags

### PR DESCRIPTION
**Pull request by atosatto**
#### What type of PR is this?

/kind feature
/sig apps

#### What this PR does / why we need it:

Remove the deprecated `--enable-taint-manager` flag in `kube-controller-manager` and cleans-up the code.

#### Which issue(s) this PR fixes:

Rebase of https://github.com/kubernetes/kubernetes/pull/112312

#### Does this PR introduce a user-facing change?

```release-note
Remove deprecated `--enable-taint-manager` and `--pod-eviction-timeout` CLI flags
```

### CODE REVIEW

**Comment by liggitt** [comment]
the mechanics of dropping the flag and config type plumbing are trivial and lgtm. It's more important to get approval of the nodelifecycle controller change from an approver in that package

File: cmd/kube-controller-manager/app/options/nodelifecyclecontroller.go [hunk 1]
```diff
@@ -44,9 +44,7 @@ func (o *NodeLifecycleControllerOptions) AddFlags(fs *pflag.FlagSet) {
...
 	fs.Float32Var(&o.UnhealthyZoneThreshold, "unhealthy-zone-threshold", 0.55, "Fraction of Nodes in a zone which needs to be not Ready (minimum 3) for zone to be treated as unhealthy. ")
-	fs.BoolVar(&o.EnableTaintManager, "enable-taint-manager", o.EnableTaintManager, "If set to true enables NoExecute Taints and will evict all not-tolerating Pod running on Nodes tainted with this kind of Taints.")
-	fs.MarkDeprecated("enable-taint-manager", "This flag is deprecated and it will be removed in 1.27. The taint-manager is enabled by default and will remain implicitly enabled once this flag is removed.")
-	fs.MarkDeprecated("pod-eviction-timeout", "This flag is deprecated and it will be removed in 1.27. Once taint manager is enabled, this flag has no effect.")
+	fs.MarkDeprecated("pod-eviction-timeout", "This flag is deprecated and it will be removed in 1.27. Since taint manager is always enabled, this flag will have no effect.")
```
Comment (pacoxu) [inline cmd/kube-controller-manager/app/options/nodelifecyclecontroller.go:47]: This should be removed or changed from 1.27 to 1.28.

**Comment by pacoxu** [comment]
`pod-eviction-timeout` is marked as deprecated in v1.26.
The release note should be updated. And this may be removed as the comment says:

File: cmd/kube-controller-manager/app/options/nodelifecyclecontroller.go (hunk 1 above)
Comment (atosatto) [reply cmd/kube-controller-manager/app/options/nodelifecyclecontroller.go:47]: > This should be removed or changed from 1.27 to 1.28.

Updated both the release notes and the code.

**Comment by pacoxu** [comment]
/lgtm
/assign   @k82cn  @smarterclayton
for `pkg/controller/nodelifecycle` approval

**Review by ravisantoshgudimetla — APPROVED** [review]
/lgtm

Thank you for working on this @atosatto 

/assign @soltysh  for approval

cc @atiratree

**Review by soltysh — APPROVED** [review]
I'm mostly ok with the controller changes, although what worries me is that after this PR even though the coverage grows from 59.7% to 61.6% some areas which were previously covered (I'm assuming that's from removing `TestMonitorNodeHealthEvictPodsWithDisruption` test) are less covered. I'd probably suggest revisiting that test and only rewrite/update it to ensure the coverage is appropriate.

**Review by Huang-Wei — COMMENTED** [review]
/approve
for scheduler changes.

**Comment by furkatgofurov7** [comment]
@atosatto hi!

K8s 1.27 Bug Triage Shadow here. I see a lot of activities happening on this PR, however FYI, with Code Freeze approaching next week on March 15th, if you would like this PR to be part of the v1.27 release, please make sure it goes in before above mentioned date. Post Code Freeze, we will only be able to accept Release Blockers and Critical issues. Thank you!

**Comment by atosatto** [comment]
@liggitt for approval, thanks! 🙏

**Pull request merged**